services/ui-src/src/components/fields/TextField.tsx

```diff
@@ -44,6 +56,30 @@ export const TextField = ({
     form.setValue(name, value, { shouldValidate: true });
   };
 
+  // if should autosave, submit field data to database on blur
+  const onBlurHandler = async (event: InputChangeEvent) => {
+    if (autosave) {
+      const { name, value } = event.target;
+      if (userIsStateUser || userIsStateRep) {
+        // check field data validity
+        const fieldDataIsValid = await form.trigger(name);
+        // if valid, use; if not, reset to default
+        const fieldValue = fieldDataIsValid ? value : defaultValue;
+
+        const reportKeys = {
+          state: state,
+          id: report?.id,
+        };
+        const dataToWrite = {
+          status: ReportStatus.IN_PROGRESS,
+          lastAlteredBy: full_name,
+          fieldData: { [name]: fieldValue },
+        };
+        await updateReport(reportKeys, dataToWrite);
```

Comment:
wondering if we need to await this. It should queue the api calls if we get multiple. Would have to check

Reply:
hmm yeah we should check on that and see. you're right that it probably doesn't make sense to await this.